### PR TITLE
page: init at 2.3.4

### PR DESCRIPTION
**Pull request by s1341**
###### Motivation for this change

Add the page utility, which is a pager using nvim's :terminal feature. 

See #82948 82948

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/tools/misc/page/default.nix [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+  };
+
+  cargoSha256 = "0z1qlz0kpj2vpg3snl51qhmdgs71d2g6bs9mjkb11j545z9b0rrf";
+
+  meta = with stdenv.lib; {
+    description = "Use neovim as pager";
+    homepage = "https://github.com/I60R/page";
+    license = stenv.lib.licenses.mit;
+    platforms = platforms.all;
+    maintainer = stdenv.lib.maintainers.s1341;
```
Comment (symphorien) [inline pkgs/tools/misc/page/default.nix]: even if you are the only maintainer this should be plural and a list.

File: pkgs/tools/misc/page/default.nix [hunk 2]
```diff
@@ -0,0 +1,23 @@
+{ stdenv, rustPlatform, fetchFromGitHub }:
+
+rustPlatform.buildRustPackage rec {
+  pname = "page";
+  version = "v2.3.2";
```
Comment (symphorien) [inline pkgs/tools/misc/page/default.nix]:
```suggestion
  version = "2.3.2";
```

File: pkgs/tools/misc/page/default.nix [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+
+  src = fetchFromGitHub {
+    owner = "I60R";
+    repo = pname;
+    rev = version;
```
Comment (symphorien) [inline pkgs/tools/misc/page/default.nix]:
```suggestion
    rev = "v${version};
```

**Review by symphorien — COMMENTED** [review]
You also need to add `page` to `pkgs/top-level/all-packages.nix`.

File: pkgs/tools/misc/page/default.nix (hunk 1 above)
Comment (s1341) [reply pkgs/tools/misc/page/default.nix]: done

File: pkgs/tools/misc/page/default.nix (hunk 2 above)
Comment (s1341) [reply pkgs/tools/misc/page/default.nix]: done

File: pkgs/tools/misc/page/default.nix (hunk 3 above)
Comment (s1341) [reply pkgs/tools/misc/page/default.nix]: done.

**Comment by s1341** [comment]
@symphorien thanks for the review. Pushed the changes.

File: pkgs/tools/misc/page/default.nix [hunk 4]
```diff
@@ -0,0 +1,23 @@
...
+  version = "2.3.2";
+
+  src = fetchFromGitHub {
+    owner = "I60R";
+    repo = pname;
+    rev = "v${version}";
+    sha256 = "1plnli0h9zcmncjn8xv1114lp4pw18hajcpmb1cbmhn43isqn3v5";
+  };
+
+  cargoSha256 = "0z1qlz0kpj2vpg3snl51qhmdgs71d2g6bs9mjkb11j545z9b0rrf";
```
Comment (jonringer) [inline pkgs/tools/misc/page/default.nix]:
```suggestion
  cargoSha256 = "0kjfyyzz29fibazvfgvgh9ds3nnf8cpsm6d91d5jy47nbn060i75";
```

```
hash mismatch in fixed-output derivation '/nix/store/vs3i7lifk5gvjjvqs3pwjl7ds5hf1kcp-page-2.3.2-vendor.tar.gz':
  wanted: sha256:0z1qlz0kpj2vpg3snl51qhmdgs71d2g6bs9mjkb11j545z9b0rrf
  got:    sha256:0kjfyyzz29fibazvfgvgh9ds3nnf8cpsm6d91d5jy47nbn060i75
```

File: pkgs/tools/misc/page/default.nix [hunk 5]
```diff
@@ -0,0 +1,23 @@
+{ lib, rustPlatform, fetchFromGitHub }:
+
+rustPlatform.buildRustPackage rec {
+  pname = "page";
+  version = "2.3.2";
+
+  src = fetchFromGitHub {
```
Comment (danieldk) [inline pkgs/tools/misc/page/default.nix]: Looking at the source repo, it seems that they also include a small program to generate shell completions. You could consider installing the completion files. `installShellFiles` is really nice for installing them:

https://github.com/NixOS/nixpkgs/blob/master/pkgs/build-support/setup-hooks/install-shell-files.sh

**Comment by symphorien** [comment]
When running
```
echo test | LANG=C page
```
I get a spurious message:
```
/nix/store/kp60cyq5iig35gzp7l2pr0ilxxa4m919-bash-interactive-4.4-p23/bin/bash: warning: here-document at line 0 delimited by end-of-file (wanted `EOF')
```

File: pkgs/tools/misc/page/default.nix (hunk 5 above)
Comment (s1341) [reply pkgs/tools/misc/page/default.nix]: thanks for the input. I added this in fecca1c

**Comment by s1341** [comment]
> When running
> 
> ```
> echo test | LANG=C page
> ```
> 
> I get a spurious message:
> 
> ```
> /nix/store/kp60cyq5iig35gzp7l2pr0ilxxa4m919-bash-interactive-4.4-p23/bin/bash: warning: here-document at line 0 delimited by end-of-file (wanted `EOF')
> ```

I don't get that error from either bash or zsh...

**Comment by symphorien** [comment]
I still get it with your latest commit, in the shell given by nix-review.

**Comment by s1341** [comment]
@symphorien can you give me the full command to execute that shell?

**Comment by symphorien** [comment]
`nix-review pr 82949`

**Comment by s1341** [comment]
I see the issue... Not sure what is causing it...

**Comment by I60R** [comment]
This message appears when `$SHELL` is set to `/bin/bash` and then `:term tail -f <<EOF` is executed in neovim by page.

I'm able to reproduce it with `/bin/bash -c "tail -f <<EOF"`.

There's probably some bug on bash side and they should fix it. However, it shouldn't be blocker for us, since I simply need to find a way to create empty :term buffer even without spawning bash.

**Comment by symphorien** [comment]
Only /bin/sh exists on NixOS (it is more or less guaranteed by posix).

**Comment by I60R** [comment]
Ok, this should be fixed in the latest version

**Comment by s1341** [comment]
@symphorien, @I60R has fixed the issue you reported. I've bumped this up to 2.3.4 which includes the fix.

**Review by symphorien — APPROVED** [review]

**Review by jonringer — APPROVED** [review]
LGTM
shows usage

```
https://github.com/NixOS/nixpkgs/pull/82949
1 package built:
page
```
```
[nix-shell:/home/jon/.cache/nixpkgs-review/pr-82949]$ tree ./results/page
./results/page
├── bin
│   └── page
└── share
    ├── bash-completion
    │   └── completions
    │       └── page.bash
    ├── fish
    │   └── vendor_completions.d
    │       └── page.fish
    └── zsh
        └── site-functions
            └── _page

8 directories, 4 files
```

**Comment by jonringer** [comment]
@GrahamcOfBorg build page

**Comment by s1341** [comment]
thanks @jonringer!

**Comment by s1341** [comment]
@I60R see #205528

**Pull request merged**